components/kubeflow/pytorch-launcher/sample.py

```diff
@@ -1,36 +1,98 @@
-import copy
 import json
+from typing import NamedTuple
+from collections import namedtuple
 import kfp
-from kfp import components
 import kfp.dsl as dsl
+from kfp import components
+from kfp.dsl.types import Integer
 
 
 def get_current_namespace():
     """Returns current namespace if available, else kubeflow"""
     try:
-        current_namespace = open("/var/run/secrets/kubernetes.io/serviceaccount/namespace").read()
+        current_namespace = open(
+            "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+        ).read()
     except:
         current_namespace = "kubeflow"
     return current_namespace
 
 
-pytorchjob_launcher_op = components.load_component_from_file("./component.yaml")
-# pytorchjob_launcher_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/master/components/kubeflow/pytorch-launcher/component.yaml')
+def create_worker_spec(
+    worker_num: int = 0
+) -> NamedTuple(
+    "CreatWorkerSpec", [("worker_spec", dict)]
+):
+    """
+    Creates pytorch-job worker spec
+    """
+    worker = {}
+    if worker_num > 0:
+        worker = {
+            "replicas": worker_num,
+            "restartPolicy": "OnFailure",
+            "template": {
+                "metadata": {
+                    "annotations": {
+                        "sidecar.istio.io/inject": "false"
+                    }
+                },
+                "spec": {
+                    "containers": [
+                        {
+                            "args": [
+                                "--backend",
+                                "gloo",
+                            ],
+                            "image": "gcr.io/kubeflow-examples/pytorch-dist-mnist:v20180702-a57993c",
+                            "name": "pytorch",
+                            "resources": {
+                                "requests": {
+                                    "memory": "4Gi",
+                                    "cpu": "2000m",
+                                    # Uncomment for GPU
+                                    # "nvidia.com/gpu": 1,
+                                },
+                                "limits": {
+                                    "memory": "4Gi",
+                                    "cpu": "2000m",
+                                    # Uncomment for GPU
+                                    # "nvidia.com/gpu": 1,
+                                },
+                            },
+                        }
+                    ]
+                },
+            },
+        }
+
+    worker_spec_output = namedtuple(
+        "MyWorkerOutput", ["worker_spec"]
+    )
+    return worker_spec_output(worker)
+
+
+worker_spec_op = components.func_to_container_op(
+    create_worker_spec,
+    base_image="tensorflow/tensorflow:1.11.0-py3",
```

Comment:
Why do we need tensorflow as the base image here?

Reply:
I guess simple python image would suffice. I will update it.